sql/core/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/RowLevelOperationRuntimeGroupFiltering.scala

```diff
@@ -51,7 +51,8 @@ class RowLevelOperationRuntimeGroupFiltering(optimizeSubqueries: Rule[LogicalPla
     // apply special dynamic filtering only for group-based row-level operations
     case GroupBasedRowLevelOperation(replaceData, _, Some(cond),
         DataSourceV2ScanRelation(_, scan: SupportsRuntimeV2Filtering, _, _, _))
-        if conf.runtimeRowLevelOperationGroupFilterEnabled && cond != TrueLiteral =>
+        if conf.runtimeRowLevelOperationGroupFilterEnabled && cond != TrueLiteral
+          && scan.filterAttributes().nonEmpty =>
```

Comment:
can you elaborate a bit more about how an invalid filter can be generated?

Reply:
will cause `buildKeys` and `pruningKeys` to be empty.

```
val buildKeys = V2ExpressionUtils.resolveRefs[Attribute](filterAttrs, matchingRowsPlan)
val pruningKeys = V2ExpressionUtils.resolveRefs[Attribute](filterAttrs, r)
val dynamicPruningCond = buildDynamicPruningCond(matchingRowsPlan, buildKeys, pruningKeys)
```

Reply:
before this change, newly added test cases would fail.

error:

```
head of empty ArraySeq
java.util.NoSuchElementException: head of empty ArraySeq
	at scala.collection.IndexedSeqOps.head(IndexedSeq.scala:99)
	at scala.collection.IndexedSeqOps.head$(IndexedSeq.scala:94)
	at scala.collection.immutable.ArraySeq.head(ArraySeq.scala:35)
	at org.apache.spark.sql.catalyst.expressions.ListQuery.dataType(subquery.scala:371)
	at org.apache.spark.sql.catalyst.expressions.InSubquery.checkInputDataTypes(predicates.scala:382)
	at org.apache.spark.sql.catalyst.expressions.Expression.resolved$lzycompute(Expression.scala:261)
	at org.apache.spark.sql.catalyst.expressions.Expression.resolved(Expression.scala:261)
	at org.apache.spark.sql.catalyst.expressions.Expression.$anonfun$childrenResolved$1(Expression.scala:273)
	at org.apache.spark.sql.catalyst.expressions.Expression.$anonfun$childrenResolved$1$adapted(Expression.scala:273)
```